library/alloc/src/string.rs

```diff
@@ -120,7 +120,7 @@ use crate::vec::Vec;
 /// `String`s are always valid UTF-8. This has a few implications, the first of
 /// which is that if you need a non-UTF-8 string, consider [`OsString`]. It is
 /// similar, but without the UTF-8 constraint. The second implication is that
-/// you cannot index into a `String`:
+/// you cannot index into a `String` with a `usize`:
```

Comment:
Or, really, any other non-range value. Or maybe this is correct and it's just the language that's confusing - you can't index but you can substring slice. I didn't think of them as different terms until I started working on this PR so maybe this is good as is :sweat_smile: